### DISCUSSION TITLE
Live update wire:model.live doesn't work

### DISCUSSION THREAD

**Discussion by vincentdev15** [livewire/livewire#6431] [7. Bugs (archived)]
### Livewire version

3

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

I have create a very simple component with an input field and I display the content of the input aside the field.

```
class InputComponent extends Component
{
    public $message;

    public function render()
    {
        return view('livewire.input-component');
    }
}
```

And here is the view.

```
<div>
    <input wire:model.live="message">

    <div>{{ $message }}</div>
</div>
```

I use this component inside a blade view.

```
@livewire('input-component')
```

When I type inside the input, the message doesn't appear inside the div.

### Additional information

Today I have checked the console when I type inside the input field and I get this error.

```
Uncaught TypeError: window.Alpine.cloneNode is not a function
```

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by PhiloNL** [comment]
This seems to work just fine:

```php
class InputComponent extends Component
{
    public $message;

    public function render()
    {
        return <<<'HTML'
            <div>
                <input wire:model.live="message">
                <div>{{ $message }}</div>
            </div>
        HTML;
    }
}
```

Can you please create and share a demo repository that replicates the problem?

**Reply by vincentdev15** [reply]
I have tested by creation a fresh new Laravel application and I don't have any problem to get it work.

I have updated my post by adding the console error I've just noticed a few minutes ago.

I have found why it doesn't work.

Alpine is only available on views where a Livewire component is.

So I have decided to install AlpineJS with npm and import the AlpineJS object in the app.js file. If I remove this line, live update works. If a keep this line, live update doesn't work.

Reading the Livewire documentation once again, it's said that if I need AlpineJS in some views, because without installing it I can't get it work, even if I had installed Livewire.

Today I don't understand why but it works without any problem and I don't understand why.

I come back here if I have further information on this strange behavior.

**Reply by ahinkle** [reply]
Do you have an installed version of Alpine(via NPM)? It could be outdated. Livewire v3 includes Alpine (with cloneNode).

**Comment by pdaether** [comment]
I have exactly the same problem.

PHP: 8.2.9
Laravel: 10.19.0
Livewire: v3.0.0-beta.9

Here I don't use volt, but for the completeness it is installed as well: v1.0.0-beta.6

```php
<?php

namespace App\Livewire;

use Illuminate\Support\Facades\Log;
use Livewire\Component;

class Test extends Component
{
    public $search = '';
    public function render()
    {
        Log::warning($this->search);
        return view('livewire.test');
    }
}
```

```php
<div>
    <input wire:model.live="search" type="text" placeholder="Search" />
    <p>Search: {{ $search }}</p>
</div>
```

The render function gets executed and the correct value gets logged, but in the view the `{{ $search }}` is still empty.

**Comment by sayedsadat344** [comment]
Isn't Livewire V3 supposed to have AlpineJS with it after installation?

**Comment by alexanderabaya** [comment]
I'm having a similar issue too..

I have a `wire:model.live` inside my livewire component
`<input type="text" class="form-control" wire:model.live='exampleInput'>`

then i have a function a simple `wire:click` function 

`<button type="button" wire:click='testButton'>Test</button>`

`
    public function testButton()
    {   
        $this->exampleInput = 'Write something on input';
    }
`

When I click the button the `$exampleInput` update the values when i call it to the blade like `{{$exampleInput}}`,  but the wire model itself is not updating like it should be.

**Reply by bruceleedrupal** [reply]
Yes,I have the same problem,this do not happen in livewire version 2

**Reply by Yogabayu** [reply]
anyone have solution ?

**Reply by alexanderabaya** [reply]
If you are using livewire 3, this problem occurs when you import the Alpine JS in ``resources/js/app.js`` like this:
```
//This is livewire 2 approach
import Alpine from 'alpinejs'
import customPlugin from './plugins/custom-plugin'

Alpine.plugin(customPlugin)
Alpine.start()
```
This problem can be solved by removing alpinejs imports in ``resources/js/app.js``.

The livewire 3 documentation said  **Livewire includes Alpine and many popular Alpine plugins by default.** If you wish to add more alpine js plugins or you just want to add plugins you need (to be sure the plugin is installed incase it is not included in "defaults") you can continue reading documentation in this section [https://livewire.laravel.com/docs/upgrading#including-via-js-bundle](url)

**Reply by Yogabayu** [reply]
i don't have AlphineJS on `resource/js/app.js`, its only contain `import './bootstrap';

but if i remove my js code , its work fine. 

this is my js code : 
`
'use strict';

(function ($) {

    /*------------------
        Preloader
    --------------------*/
    $(window).on('load', function () {
        $(".loader").fadeOut();
        $("#preloder").delay(100).fadeOut("slow");

        /*------------------
            Gallery filter
        --------------------*/
        $('.featured__controls li').on('click', function () {
            $('.featured__controls li').removeClass('active');
            $(this).addClass('active');
        });
        if ($('.featured__filter').length > 0) {
            var containerEl = document.querySelector('.featured__filter');
            var mixer = mixitup(containerEl);
        }
    });

    /*------------------
        Background Set
    --------------------*/
    $('.set-bg').each(function () {
        var bg = $(this).data('setbg');
        $(this).css('background-image', 'url(' + bg + ')');
    });

    //Humberger Menu
    $(".humberger__open").on('click', function () {
        $(".humberger__menu__wrapper").addClass("show__humberger__menu__wrapper");
        $(".humberger__menu__overlay").addClass("active");
        $("body").addClass("over_hid");
    });

    $(".humberger__menu__overlay").on('click', function () {
        $(".humberger__menu__wrapper").removeClass("show__humberger__menu__wrapper");
        $(".humberger__menu__overlay").removeClass("active");
        $("body").removeClass("over_hid");
    });

    /*------------------
		Navigation
	--------------------*/
    $(".mobile-menu").slicknav({
        prependTo: '#mobile-menu-wrap',
        allowParentLinks: true
    });

    /*-----------------------
        Categories Slider
    ------------------------*/
    $(".categories__slider").owlCarousel({
        loop: true,
        margin: 0,
        items: 4,
        dots: false,
        nav: true,
        navText: ["<span class='fa fa-angle-left'><span/>", "<span class='fa fa-angle-right'><span/>"],
        animateOut: 'fadeOut',
        animateIn: 'fadeIn',
        smartSpeed: 1200,
        autoHeight: false,
        autoplay: true,
        responsive: {

            0: {
                items: 1,
            },

            480: {
                items: 2,
            },

            768: {
                items: 3,
            },

            992: {
                items: 4,
            }
        }
    });


    $('.hero__categories__all').on('click', function(){
        $('.hero__categories ul').slideToggle(400);
    });

    /*--------------------------
        Latest Product Slider
    ----------------------------*/
    $(".latest-product__slider").owlCarousel({
        loop: true,
        margin: 0,
        items: 1,
        dots: false,
        nav: true,
        navText: ["<span class='fa fa-angle-left'><span/>", "<span class='fa fa-angle-right'><span/>"],
        smartSpeed: 5000,
        autoHeight: false,
        autoplay: true
    });

    /*-----------------------------
        Product Discount Slider
    -------------------------------*/
    $(".product__discount__slider").owlCarousel({
        loop: true,
        margin: 0,
        items: 3,
        dots: true,
        smartSpeed: 1200,
        autoHeight: false,
        autoplay: true,
        responsive: {

            320: {
                items: 1,
            },

            480: {
                items: 2,
            },

            768: {
                items: 2,
            },

            992: {
                items: 3,
            }
        }
    });

    /*---------------------------------
        Product Details Pic Slider
    ----------------------------------*/
    $(".product__details__pic__slider").owlCarousel({
        loop: true,
        margin: 20,
        items: 4,
        dots: true,
        smartSpeed: 1200,
        autoHeight: false,
        autoplay: true
    });

    /*-----------------------
		Price Range Slider
	------------------------ */
    var rangeSlider = $(".price-range"),
        minamount = $("#minamount"),
        maxamount = $("#maxamount"),
        minPrice = rangeSlider.data('min'),
        maxPrice = rangeSlider.data('max');
    rangeSlider.slider({
        range: true,
        min: minPrice,
        max: maxPrice,
        values: [minPrice, maxPrice],
        slide: function (event, ui) {
            minamount.val('Rp' + ui.values[0]);
            maxamount.val('Rp' + ui.values[1]);
        }
    });
    minamount.val('Rp' + rangeSlider.slider("values", 0));
    maxamount.val('Rp' + rangeSlider.slider("values", 1));

    /*--------------------------
        Select
    ----------------------------*/
    $("select").niceSelect();

    /*------------------
		Single Product
	--------------------*/
    $('.product__details__pic__slider img').on('click', function () {

        var imgurl = $(this).data('imgbigurl');
        var bigImg = $('.product__details__pic__item--large').attr('src');
        if (imgurl != bigImg) {
            $('.product__details__pic__item--large').attr({
                src: imgurl
            });
        }
    });

    /*-------------------
		Quantity change
	--------------------- */
    var proQty = $('.pro-qty');
    proQty.prepend('<span class="dec qtybtn">-</span>');
    proQty.append('<span class="inc qtybtn">+</span>');
    proQty.on('click', '.qtybtn', function () {
        var $button = $(this);
        var oldValue = $button.parent().find('input').val();
        if ($button.hasClass('inc')) {
            var newVal = parseFloat(oldValue) + 1;
        } else {
            // Don't allow decrementing below zero
            if (oldValue > 0) {
                var newVal = parseFloat(oldValue) - 1;
            } else {
                newVal = 0;
            }
        }
        $button.parent().find('input').val(newVal);
    });

})(jQuery);`

`

**Reply by HM-Admin** [reply]
You solved my issue, removing alpinejs from app.js solved the issue updating form inputs.

Thanks

**Reply by PhillipMwaniki** [reply]
@Yogabayu did you get a fix for your issue? I have a similar problem.

**Reply by Yogabayu** [reply]
> @Yogabayu did you get a fix for your issue? I have a similar problem.

I renamed my JS file, I don't know why but then my LiveWire started working normally. The initial name of my js file was wrong, main.js, I changed it to main1.js

**Reply by DilieCat** [reply]
@PhillipMwaniki Were you able to fix it? I don't have any JS file like Yogabayu just a simple Jetstream project.

**Reply by SergeyChechel** [reply]
https://livewire.laravel.com/docs/upgrading#wiremodel

**Comment by BarionTechnology** [comment]
I have a similar problem, when I type some work in my search if I'm not logged, livewire redirect me to login. I also using livewire 3

**Reply by SergeyChechel** [reply]
https://livewire.laravel.com/docs/upgrading#wiremodel

**Comment by pauledenburg** [comment]
I got it fixed.
I had the exact same problem.

Using an input field with a model attached would not live update the text on the page.

Livewire 3 does no live updates by default like v2 did. See: https://livewire.laravel.com/docs/upgrading#wiremodel 

```
<!-- turn this: 'wire:model' -->
<input type="text" wire:model="title" />
<h1>{{$title}}</h1>

<!-- into this: 'wire:model.live' -->
<input type="text" wire:model.live="title" />
<h1>{{$title}}</h1>
```

**Reply by Bakugo90** [reply]
it work for me too. Thank you @pauledenburg

**Comment by HM-Admin** [comment]


**Reply by joshhanley** [reply]
@HM-Admin can you please stop spamming this thread? Thanks!

**Comment by HM-Admin** [comment]
Estimado livewire/livewire,

Su ticket de soporte "Re: [livewire/livewire] Live update wire:model.live doesn't work (Dis" ha sido enviado.

Respondemos a todos los tickets lo antes posible, en un plazo de 24 a 48 horas. Si prevemos que su ticket tardará más tiempo, le informaremos enviándole un correo electrónico.

ID de seguimiento del ticket: TJ2-U4G-QWPL

Puede ver el estado de su ticket aquí:
http://soporte.hellomove.es/ticket.php?track=TJ2-U4G-QWPL&e=reply%2BA5HE3KRV3NHE2G5OQXZRZKGEZNWYBEVBMXHAAVEMMU%40reply.github.com&Refresh=82978 

Recibirá una notificación por correo electrónico cuando nuestro personal responda a su ticket.


Atentamente,

Soporte
http://soporte.hellomove.es 

--b1=_pnz859u52waBKf65NeqYfOa4QAOtk431RuAz7OnTM7Q
Content-Type: text/html; charset=UTF-8
Content-Transfer-Encoding: 8bit

<p>Estimado livewire/livewire,</p>
<p>Su ticket de soporte "Re: [livewire/livewire] Live update wire:model.live doesn't work (Dis" se ha enviado.</p>
<p>Respondemos a todas las entradas lo antes posible, en un plazo de 24 a 48 horas. Si prevemos que su solicitud tardará más tiempo, le informaremos enviándole un correo electrónico..</p>
<p>ID de seguimiento del ticket: TJ2-U4G-QWPL</p>
<p>Puede ver el estado de su ticket aquí:<br /><a href="http://soporte.hellomove.es/ticket.php?track=TJ2-U4G-QWPL&amp;e=reply%2BA5HE3KRV3NHE2G5OQXZRZKGEZNWYBEVBMXHAAVEMMU%40reply.github.com&amp;Refresh=82978">http://soporte.hellomove.es/ticket.php?track=TJ2-U4G-QWPL&amp;e=reply%2BA5HE3KRV3NHE2G5OQXZRZKGEZNWYBEVBMXHAAVEMMU%40reply.github.com&amp;Refresh=82978</a></p>
<p>Recibirá una notificación por correo electrónico cuando nuestro personal responda a su ticket.</p>
<p>Atentamente,</p>
<p>Soporte<br /><a href="http://soporte.hellomove.es">http://soporte.hellomove.es</a></p>


--b1=_pnz859u52waBKf65NeqYfOa4QAOtk431RuAz7OnTM7Q--
